main.go

```diff
@@ -173,6 +173,22 @@ func (m app) waitForSignals() {
 }
 
 func (m app) teardown() {
-	m.queue.stop()
-	m.consumerTeardown.Wait()
+	var consumerTeardown sync.WaitGroup
+	m.queue.stop(consumerTeardown)
```

Comment:
does this work? don't you need to send `&consumerTeardown` to the stop method? I think if you don't send the reference, the method will receive a copy of the `WaitGroup`.